src/main/java/com/m2m/shopping/Service/Impl/ProductServiceImpl.java

```diff
@@ -31,4 +30,72 @@ public Product save(Product enity) {
     public boolean delete(Product enity) {
         return super.delete(enity);
     }
+
+    @Override
+    public List<Product> findToPage(Boolean existActive, int pageNumber, int pageSize) {
+        return super.findToPage(Product.class,existActive,pageNumber,pageSize);
+    }
+
+    @Override
+    public List<Product> findPageParam( Boolean existActive, int pageNumber, int pageSize, String title, Integer idCategories) {
+        String sql = "select o from Product o where o.title like ?0 and o.categories.id = ?1";
+        return super.findManyPage(Product.class,sql,pageNumber,pageSize,"%"+title+"%", idCategories );
```

Comment:
@truongnguyenfpthcm2605 

Param `existActive` không có dùng ở đâu trong hàm này, nếu không dùng thì nên loại ra nha em.

Không được có biến dư thừa khi code.

Reply:
@Override
    public List<Product> findPageParam(Boolean existActive, int pageNumber, int pageSize, String title, Integer idCategories) {
        String sql = "select o from Product o where o.isActive = ?0 and  o.title like ?1 and o.categories.id = ?2";
        return super.findManyPage(Product.class, sql, pageNumber, pageSize,existActive, "%"+title+"%", idCategories );
    }
    Em đã fix lại phần Param existActive và cách format code